protocols/kad/tests/client_mode.rs

```diff
@@ -172,10 +172,10 @@ impl MyBehaviour {
         let local_peer_id = k.public().to_peer_id();
 
         Self {
-            identify: identify::Behaviour::new(identify::Config::new(
-                "/test/1.0.0".to_owned(),
-                k.public(),
-            )),
+            identify: identify::Behaviour::new(
+                identify::Config::new("/test/1.0.0".to_owned(), k.public())
+                    .with_push_external_addr_updates(false),
+            ),
```

Comment:
Do you know why this test requires `with_push_external_addr_upates(false)`?

Reply:
The test expects certain number of events, with this change there is a chance of one more push to happen from `Identify`, but due to async nature sometimes it arrives sooner and sometimes later.

Fails like this:
```
thread 'adding_an_external_addresses_activates_server_mode_on_existing_connections' panicked at ../rust-libp2p/swarm-test/src/lib.rs:170:13:
expected 2 items from second swarm but got 3
```

Reply:
Yeah this makes sense. I am okay with disabling this in this test.